### ISSUE TITLE
Bug when using tfa.seq2seq.AttentionWrapper as input to tf.keras.layers.Dense

### ISSUE THREAD

**Issue by samuel-lunii** [tensorflow/addons#1556]
**System information**
- OS Platform and Distribution : Debian 4.9.168-1+deb9u5 (2019-08-11) x86_64
- TensorFlow version and how it was installed (source or binary): pip install tensorflow == 2.1.0
- TensorFlow-Addons version and how it was installed (source or binary): 0.8.3, pip install tensorflow_addons
- Python version: 3.5.3
- Is GPU used? (yes/no): no

**Describe the bug**

I use tfa.seq2seq.AttentionWrapper as an input to tf.keras.layers.Dense and get an `AttributeError: 'AttentionWrapper' object has no attribute 'shape'` 

**Code to reproduce the issue**

```
import tensorflow as tf
import tensorflow_addons as tfa

inputs = tf.random.uniform((128, 1200), maxval = 24, dtype = tf.int32)

# Attention
attention_cell = tfa.seq2seq.AttentionWrapper(
    cell = tf.keras.layers.GRUCell(units = 256), 
    attention_layer_size = 256,
    attention_mechanism = tfa.seq2seq.BahdanauAttention(256, inputs),
    alignment_history=True,
    output_attention=False
)     

tf.keras.layers.Dense(256)(attention_cell)
```

**Other info / logs**

```
Traceback (most recent call last):
  File "reproduce_attention_error.py", line 15, in <module>
    tf.keras.layers.Dense(256)(attention_cell)
  File "/home/delalez/.local/lib/python3.5/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 818, in __call__
    self._maybe_build(inputs)
  File "/home/delalez/.local/lib/python3.5/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 2098, in _maybe_build
    self.input_spec, inputs, self.name)
  File "/home/delalez/.local/lib/python3.5/site-packages/tensorflow_core/python/keras/engine/input_spec.py", line 163, in assert_input_compatibility
    if x.shape.ndims is None:
AttributeError: 'AttentionWrapper' object has no attribute 'shape'
```

**Comment by guillaumekln** [comment]
What are you trying to achieve exactly? There are 2 issues with your code:

* `attention_cell` is an RNN cell and should first be called with an input and a state
* the `inputs` your are passing to the attention mechanism should have the shape `[batch, time, depth]`

**Comment by samuel-lunii** [comment]
Thanks for quick reply ! 

I'm trying to migrate a code from TF1.14 to TF2.1.

I wrote the line `tf.keras.layers.Dense(256)(attention_cell)` because I tried to replace `tf.contrib.rnn.OutputProjectionWrapper(attention_cell, 256)` from the original code. 

I read [here](https://github.com/ageron/handson-ml/issues/429) that the `OutputProjectionWrapper` could be replaced with a `Dense` layer. But now I realize that i got it wrong ! 

I'll try [this](https://stackoverflow.com/questions/57482257/how-to-replace-outputprojectionwrapper-in-tensforflow-2-0) and will let you know if it solved the issue.
